pynitrokey/cli/fido2.py

```diff
@@ -191,6 +191,114 @@ def list():
         local_print(f"{id_}: {name}")
 
 
+@click.command()
+@click.option(
+    "-s",
+    "--serial",
+    help="Serial number of Nitrokey to use. Prefix with 'device=' to provide device file, e.g. 'device=/dev/hidraw5'.",
+)
+@click.option("--pin", help="provide PIN instead of asking the user", default=None)
+def list_credentials(serial, pin):
+    """List all credentials saved on the key as well as the amount of remaining slots."""
+
+    # Gets device and client_pin object
+    device = nkfido2.find(serial)
+    client_pin = ClientPin(device.ctap2)
+
+    # Makes sure pin exists
+    if not pin:
+        pin = AskUser.hidden("Please provide pin: ")
+
+    try:
+        client_token = client_pin.get_pin_token(pin)
+    except Exception as e:
+        if "PIN_NOT_SET" in str(e):
+            local_critical("Please set a pin in order to manage credentials")
+
+    cred_manager = CredentialManagement(device.ctap2, client_pin.protocol, client_token)
```

Comment:
I just found out that we already have a `cred_mgmt` function in the client.  Can we use it here to reduce code duplication?

https://github.com/Nitrokey/pynitrokey/blob/6c9d81f5d9d088b8b56c5f932736885b65fd6e05/pynitrokey/fido2/client.py#L340-L349

Reply:
We can and now are